### ISSUE TITLE
Indexing issues

### ISSUE THREAD

**Issue by melish** [EEA-Norway-Grants/dataviz#329]
Example 1:

a programme is funded by   both FM but the filter is applied to EEA grants. The programme areas from   both FM are shown in the list.
--

Example 2:
an organisation is both an   NFP and PO and the filter is applied to the PO. Still the NFP role and the   corresponding programmes are listed for the organisation.
--

**Comment by melish** [comment]
The underlying issue is that Iceland and Liechtenstein need to be able to filter on only the EEA FM and trust that what then comes up is not funded by the Norweginan FM. They don't want to wrongly take credit for something that they are not funding.
The FM filter is very special in this regards and that is also why we suggested that it is a checkbox not a drop down.
We also need the relationship between the FM/PS/PA like is in the other scenarios, meaning for example when I have selected EEA Grants I shouldn't be able to select PS Green Industry innovation as it's under the N FM. The PS can however still shot up under the PL04 programme text, that fine because that text says that it funded by both FM.


It's fine that the text explains all the different roles of the organisation even if we have filtered on just one of those roles. In this case, we should however still make sure the the FM filter applies. For example when I filter on EEA Grants, and donor project partner role the "Norwegian Institute for Air Research" should not list the projects that are under the Norwegian FM such as the CZ09 projects or alternatively list which ones are funded by each of the FM. This goes back to the underlying issue described above.
